dynamic_approach.py

```diff
@@ -0,0 +1,33 @@
+def count_expression_powers(expression, number):
+    expression_powers = ['1']
+    number_digits_amount = 0
+    i = 1
+    while len(expression) >= number_digits_amount:
+        number_in_power = int(number) ** i
+        binary_number = bin(int(number_in_power))[2:]
+        number_digits_amount = len(binary_number)
+        expression_powers.append(binary_number)
+        i += 1
+    expression_powers.pop()
+    return expression_powers
+
+
+def count_min_slices(expression, number):
+    expression_powers = count_expression_powers(expression, number)
+    slices = []
+    while True:
```

Comment:
The statement while (true) is not a good practice because it makes it hard to maintain and the way you are escaping the loop is considered bad practice.
Try to use instead:
do {
  //do something
} while (condition);